fastify.js

```diff
@@ -82,7 +84,7 @@ function build (options) {
     versioning: options.versioning
   })
   // 404 router, used for handling encapsulated 404 handlers
-  const fourOhFour = FindMyWay({ defaultRoute: fourOhFourFallBack })
+  const fourOhFour = fourOhFourManager(logger, modifyCoreObjects, genReqId)
```

Comment:
I think it is better to pass the entire `options` object, so you don't need to update this code in the future. *(in the same way we did for `createLogger` and `createServer`)*

Reply:
I have looked how to do it but I don't figure out a good way because:

+ logger is built and referenced in the logger const. In this case, I should repeat the `createLogger(options)` inside the fourOhFour file so I don't think is the wanted behaviour right?
+ modifyCoreObjects also this const should I replicate the `options.modifyCoreObjects !== false` condition?
+ also for genReqId  where there is the factory `options.genReqId || reqIdGenFactory(requestIdHeader)`

I'm not sure ether to change the options object because it can widely use now (with the 'freeze feature')

Reply:
the options are cloned, so it should not be an issue.

Reply:
Done.
Please take a look if I have understood right.
Thank you